etcdserver/api/v3rpc/watch.go

```diff
@@ -336,7 +340,36 @@ func (sws *serverWatchStream) sendLoop() {
 			}
 
 			mvcc.ReportEventReceived(len(evs))
-			if err := sws.gRPCStream.Send(wr); err != nil {
+			if !sws.fragmentWatchResponse[wresp.WatchID] {
+				if err := sws.gRPCStream.Send(wr); err != nil {
+					return
+				}
+				continue
+			}
+			var sendFragments func(maxEventsPerMsg int, wr *pb.WatchResponse) error
+			sendFragments = func(maxEventsPerMsg int, wr *pb.WatchResponse) error {
```

Comment:
maybe this should be `func (sws *serverWatchStream) sendFragments(maxEvents int, wr *pb.WatchResponse) error`